site/docs/view-spec.md

```diff
@@ -0,0 +1,253 @@
+* Support versioning of views to track how a view evolved over time
+
+## Proposal
+
+### Storage Format
+
+The view metadata storage and retrieval mirrors how Iceberg table metadata is stored and retrieved. The view metadata is stored in a JSON file on S3 for ease of tracking the evolution of the view. Metastore continues to hold the view object with some properties such as database name, owner, create time, last access time and an indication that the object is a view. 
+
+Each ‘CREATE OR REPLACE VIEW’ statement creates a new view version metadata file for that view. The view version metadata file has following fields:
+
+
+| Required/Optional | Field Name | Description |
+|-------------------|------------|-------------|
+| Required | format-version | Json format version number for the view metadata spec. The view metadata spec and the corresponding format-version is independent of table spec. Starts with 1 and is incremented when there is a breaking change to view metadata. |
+| Required | object-type    | String identifying the type of object this metadata file is for. ‘View’ for all objects covered in this spec. |
+| Required | location | Location of the view metadata files |
+| Required | current-version-id | Current / latest version of the view. Set to ‘1’ when the view is first created. |
+| Optional | properties | A string to string map of view properties. Contains pre-set properties such as ‘comment’ describing the view, does not contain arbitrary metadata. |
+| Required | versions | An array of structs describing the last few versions of the view. Controlled by the table property: “version.history.num_entries”. See more below. |
+| Required | version-log | An array of structs describing the log of created versions. See more below. |
+| Optional | schemas | A list of schemas and alias-comment maps stored as objects with schema-id. See more below |
+| Optional | current-schema-id | ID of the current schema of the view |
+
+
+‘schemas’ is an array of structs with a field as shown below:
+
+| Required/Optional | Field Name | Description |
+|-------------------|------------|-------------|
+| Optional | schema | An array of structs describing the schema of the view. This field is the same as the ‘schema’ field from Iceberg table spec. See more below. |
+
+
+‘Versions’ is an array of structs with fields as shown below:
+
+| Required/Optional | Field Name | Description |
+|-------------------|------------|-------------|
+| Required | version-id | Monotonically increasing id indicating the version of the view. Starts with “1”. |
+| Required | timestamp-ms | Timestamp expressed in ms since epoch at which the version of the view was created. |
+| Required | summary | A string to string map of view properties to track version metadata. This field can be used by engines to store any necessary properties. Two currently required properties are described below. |
+| Required | sql | A string representing SQL definition of the view as input |
```

Comment:
Given the development of Substrait, I feel we can use a more generic word (like "content", "representation"?), and then in the future we can make "substrait" a dialect? Would that work? @jacques-n

Reply:
@jacques-n How about this trade-off?
- Support a list of representations
- Types should be well defined
- Only allow one object per type in the list
- Only define one type "OriginalSql" in this spec
- Need to go through future discussions to add more types

Reply:
I agree @jzhuge that we shouldn't define a ton of representations up front. Though I am not sure what the downside of @jacques-n proposal is. Specifically: adding `sql` type and `dialect`. It is likely that all initial implementations would only contain 1 element in the list and only the dialect that the view was created by. But thats ok, we could add features to extend that list later: dialect translation or substrait etc.

I am not sure what the `OriginalSql` above would be? To me we have to specify a dialect so others know if they can read the sql or not right?